lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -159,14 +171,20 @@ def run(verify_result: true, allow_protected_ns: false, prune: true)
         @logger.summary.add_paragraph(ColorizedString.new(warning).yellow)
         success = true
       end
-    rescue FatalDeploymentError => error
+      ::StatsD.measure('all_resources.duration', StatsD.duration(start), tags: statsd_tags << "status:success")
+      success
+    rescue DeploymentTimeoutError => error
+      success = false
       @logger.summary.add_action(error.message)
+      ::StatsD.measure('all_resources.duration', StatsD.duration(start), tags: statsd_tags << "status:timeout")
+      raise
+    rescue FatalDeploymentError => error
+      @logger.summary.add_action(error.message) if error.message.present?
+      ::StatsD.measure('all_resources.duration', StatsD.duration(start), tags: statsd_tags << "status:failed")
       success = false
+      raise
     ensure
       @logger.print_summary(success)
```

Comment:
Did you miss this comment? https://github.com/Shopify/kubernetes-deploy/pull/244#discussion_r174644083
I'm not married to that specific implementation, but the problem described is still an issue in this version. I realized the other day that getting rid of the `ensure` would have the nice side-effect of not printing the result banner in the case of unexpected exceptions that are dumping backtraces.

Reply:
I misunderstood what you were suggesting since `print_summary` takes a bool right now. I'll fix this up.